chunk/chunk.go

```diff
@@ -119,6 +120,20 @@ func (a *Address) UnmarshalJSON(value []byte) error {
 	return nil
 }
 
+func (a *Address) UnmarshalString(s string) error {
+	v, err := hex.DecodeString(s)
+	if err != nil {
+		return err
+	}
+
+	if len(v) != AddressLength {
```

Comment:
Will this work with encrypted addresses?

Reply:
i still need to check on this :+1: